### ISSUE TITLE
Wrong conditional checkings

### ISSUE THREAD

**Issue by mpiorowski** [ota-meshi/eslint-plugin-svelte#254]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.23.1

### What version of `eslint-plugin-svelte` are you using?

2.8.0

### What did you do?

Simple true/false check.

### What did you expect to happen?

No eslitn errors.

### What actually happened?

![image](https://user-images.githubusercontent.com/26543876/190232999-b02b7538-0446-473a-8943-442966d643e1.png)

### Link to Minimal Reproducible Example

https://github.com/mpiorowski/svelte-eslint-bug/blob/main/src/routes/%2Bpage.svelte

### Additional comments

Thank You for fixing my previous issue, Your plugin is much better then the native svelte one :)

**Comment by baseballyama** [comment]
After remove all pretiier stuff, the issue is gone.
So this is not this ESLint plugin's issue.

But why you need to override `parser` in `.prettierrc`?
```json
"overrides": [{ "files": "*.svelte", "options": { "parser": "svelte-eslint-parser" } }]
```

**Comment by mpiorowski** [comment]
I have removed pretter form eslint and stil have this error @baseballyama .

And this setting is `.prettierrc` was the default one when i create new sveltekit project ;p Didn't even notice.

What excatly did You deleted to make it not throw error?

**Comment by baseballyama** [comment]
DId you do the below steps?

1. run `rm -rf ./node_modules`
2. run `npm i`
3. restart your IDE

**Comment by mpiorowski** [comment]
@baseballyama did all of it, and the eslint still throw the warning

**Comment by ota-meshi** [comment]
Probably treated and reported the same as the following typescript-eslint DEMO.

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=DYUwLgBAlgzg8gBxAOwgXggMwIbBiAbgCgjRIoATALghjACcpkBzCAHwgFdkKRMmQFdFx58BFYsSIAKaQEp0APgjTYiFMICEapMjlyp2+LogAyU9CEB6KxAA8AWgj0QCAPb0wgoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK+SpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

```ts
let isOpen = false;

let id: string | undefined = undefined;
;

(() => (isOpen = !isOpen));

!isOpen && id // <- reported
```

It might be possible to solve it by creating a temporary scope in the parser and letting typescript build the type information.


```ts
let isOpen = false;

let id: string | undefined = undefined;
;
function tmp () {
(() => (isOpen = !isOpen));

!isOpen && id // <- OK
}
```

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=DYUwLgBAlgzg8gBxAOwgXggMwIbBiAbgCgjRIoATALghjACcpkBzCAHwgFdkKRMmQFdFx58BFYsUzcAxmCgB7VGAC2CABQBKCAG8iECOq3oAfIdiIUwgIQWkyTZuL6It+PYgAyT9CEB6PwgAHgBaCDgAaSIAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK+SpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

**Comment by ota-meshi** [comment]
Hmm... I've considered it a bit, but it can be difficult.
It's difficult to create a temporary scope if it have the following script.

```ts
export let isOpen = false;
export let id: string | undefined = undefined;

$: a = !isOpen/*NG*/ && id
```

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=KYDwDg9gTgLgBAG2PAlgZwPJmAOzgXjgDMBDBNYAbgChRJZFk4UATALjjRihRwHM4AHzgBXHC2BFewFgVHjJ0ljWoASDiTkBCdFlwB6AFQA5AOKH9cAGRXmLakA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK+SpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

**Comment by mpiorowski** [comment]
Funny thing is, the typescript evalute `isOpen` variable correctly as boolean, not true / false, and still eslint throws warning ;p

**Comment by mpiorowski** [comment]
@ota-meshi and the other funny thing is, this condition works great when You use one of the variables, `isOpen` or `id` :D No warning then (logically it should also throw warning). BUT when You add the "&&" then the magic starts happening :D
